src/Form/index.tsx

```diff
@@ -395,11 +401,13 @@ function Form({
     }
   }, [stepKey]);
 
-  function addRepeatedRow() {
-    // Collect a list of all repeated elements
-    const repeatedServarFields = activeStep.servar_fields.filter(
-      (field: any) => field.servar.repeated
-    );
+  function addRepeatedRow(repeatContainer: Subgrid | undefined) {
+    // Collect a list of all relevant repeated elements.
+    // Legacy: if no repeatContainer is provided, then all repeated elements are relevant
```

Comment:
instead of leaving legacy logic in the react library, i think we should do a backend migration so all forms will now specify the repeating container. should be a backwards compatible backend change i believe

Reply:
BE migration added.  This code was then simplified since repeat_container always set.